src/app/chat/event_handling.nim

```diff
@@ -102,10 +102,8 @@ proc handleChatEvents(self: ChatController) =
 
   self.status.events.on("channelLoaded") do(e: Args):
     var channel = ChannelArgs(e)
-    if channel.chat.chatType == ChatType.Timeline:
-      self.view.setTimelineChat(channel.chat)
     # Do not add community chats to the normal chat list
-    elif channel.chat.chatType != ChatType.Profile and channel.chat.chatType != status_chat.ChatType.CommunityChat:
```

Comment:
So this something that should be double checked.. I'm not sure if we can simply remove the check for `ChatType.Profile` here. People might still be subscribed to channels that are of that type. And if they are and we don't have this check, we might end up adding those profile type channels to the channel list.

But this is just a thought, it may not apply...

We could test this by checking the following:

- in `master` (not sure if it's the case in `base_bc`) we're subscribing to profile channels during bootstrap of the app I believe. Whenever someone is added as a contact, we also subscribe to that user's channel (the channel is the public key of the users and the type is `profile`)
- So with that in mind, we remove the profile condition that is remove here in `master` as well, but leaving everything else untouched.
- If my theory is correct, then we should see profile channels being added to your chat list for every contact you have

Reply:
@PascalPrecht your assumption turns out to be correct. Should I leave this line as is?

Reply:
You could also look into us not subscribing to the profile channels in the first place. I think this happens somewhere during initialization, when we get our contacts. Removing the subscriptions would make the check for `ChatType.Profile` obsolete.

Reply:
so I removed some more parts here as well as in `status-lib` including in `vendor/status-lib/status/contacts.nim` where I removed `discard status_chat.createProfileChat(contact.id)` but I still see the weird chats :) currently I don't know or I don't know how to know where else to look for, would you have any idea of what else needs to be removed?

Reply:
@alexandraB99 Had to dig for it as well and this is the spot where stuff gets interesting:

https://github.com/status-im/status-lib/blob/cd5412823a8727e2decc3be430818309c57b3a92/status/chat.nim#L267-L290

Basically, what happens there is that we ensure that:

1. We have a timeline chat in case it doesn't exist ye
https://github.com/status-im/status-lib/blob/cd5412823a8727e2decc3be430818309c57b3a92/status/chat.nim#L269-L272
2. We check if we're subscribed to our own profile updates channel (needed to see our own profile updates in the timeline). Notice that `getTimelineId()` returns either the general timeline channel id, or the channel id of the individual timeline of a given user (pubkey). In this case, it's our own:
https://github.com/status-im/status-lib/blob/cd5412823a8727e2decc3be430818309c57b3a92/status/chat.nim#L274-L279
3. We ensure we subscribe to all profile update channels from contacts that we haven't subscribed to yet (as mentioned in the comment, this was needed for forward compatibility)
https://github.com/status-im/status-lib/blob/cd5412823a8727e2decc3be430818309c57b3a92/status/chat.nim#L281-L290
4. Last but not least, we set up filters for all of the channels we've saved so we receive their individual messages:
https://github.com/status-im/status-lib/blob/cd5412823a8727e2decc3be430818309c57b3a92/status/chat.nim#L292-L298

I wrote that code myself but didn't remember correctly how it behaves. I thought we always start with a blank slate and on every bootstrap we add/create profile update channels for all of our contacts. This is not the case. We merely fill in the gaps in case we aren't subscribed to some contact's profile update channel yet.

So the reason you're still seeing those profile update channels in your chat list, is because:

a) We no longer check for `chatType.Profile` so all chat items are added to the list (this you know already) and
b) We actually never unsubscribed from all of these channels.

This means, what we'd need to do, is kind of the reverse of what is being done in step 3).
Instead of filling the gaps and add missing profile update channels, we should remove all chats in status-go that are of type `Profile`, so when we hit step 4), we don't actually create filters for those anymore.

Lemme know if this makes sense!

Reply:
@PascalPrecht should be fixed, please have a look [here](https://github.com/status-im/status-go/pull/2451) and let me know any comments!

Reply:
@alexandraB99 Thanks! I think there was a little misunderstanding. See https://github.com/status-im/status-go/pull/2451#pullrequestreview-821271105 for more context.

I actually thought we could do something like (pseudo code):

```
if profileChatIds.len > 0: // We're still subscribed to profile channels, lets remove those!
  for chatId in profileChatIds:
    status.removeChat(chatId)
```

While this check will be run on every bootstrap of the app, the removal will only be performed once. That way we get rid of those channels without touching the protocol (and possibly affecting other clients [mobile] as well)

Reply:
done